content/en/blog/news/first-post/index.md

```diff
@@ -3,15 +3,117 @@ date: 2024-02-28
+
+This also has the benefit of moving the core security patching of web servers, queue listeners, and other application boundaries to the host, rather than image artifacts themselves. But what if you mostly rely on images provided by your cloud-provider? Or deploy to environments that are not easily reproducible?
+
+This is where [runtime-class-manager](https://github.com/spinkube/runtime-class-manager) (coming soon, formerly KWasm) comes into play - a Production Ready and Kubernetes-native way to manage WebAssembly runtimes on Kubernetes Hosts. With runtime-class-manager you can ensure that containerd-shim-spin is installed on your hosts, manage version lifecycles, and security patches - all from the Kubernetes API.
+
+By combining these technologies, scaling a WebAssembly application looks more like this:
+
+| Phase | Description |
+| --- | --- |
+| Kube init | Kubernetes schedules a Pod to a node, the node then prepares to run the Pod |
+| Image pull (cacheable) | The node pulls the needed container images from a remote repository to a local cache when needed |
+| WASM Load (cachable) | The shim loads the webassembly and prepares it for execution on the node’s CPU architecture |
+| Application start | The shim starts listening on the configured port and is ready to serve |
+| Pod ready | The Pod responds to the asynchronous readiness probe and signals that it is now ready |
+| Service ready | The Pod gets added to the list of active endpoints for a Service and can start receiving traffic |
+
+Or in practice:
+
+<video src="spinkube-scaling.mp4" width="80%" controls></video>
+
+
+## Deploying applications
+
+This leaves one major area that can be painful when deploying applications to Kubernetes today: the developer and deployment experience.
+
+This is where [Spin][spin], [the `spin kube` plugin](https://github.com/spinkube/spin-plugin-kube), and [`spin-operator`][spin-operator] come into play.
+
+![](spinkube-diagram.png)
+
+The spin-operator makes managing serverless applications easy - you provide the
+image and bindings to secrets, and the operator handles realizing that
+configuration as regular Kubernetes objects.
+
+Combining the operator with the `spin kube` plugin gives you superpowers, as it
+streamlines the process of generating Kubernetes YAML for your application and
+gives you a powerful starting point for applications.
+
+This is all it takes to get a HTTP application running on Kubernetes (after
+installing the containerd shim and the operator):
+
+```bash
+spin new -t http-rust spin-kube-app --accept-defaults
+cd spin-kube-app
+spin build
+export IMAGE_NAME=ttl.sh/spin-app-$(uuidgen):1h
+spin registry push $IMAGE_NAME
+spin kube scaffold -f $IMAGE_NAME > app.yaml
+kubectl apply -f app.yaml
```

Comment:
What about adding some inline comments like this

```bash
# Create a new Spin App
spin new -t http-rust --accept-defaults spin-kube-app
cd spin-kube-app

# Build the Spin App
spin build

# Push the Spin App to an OCI compliant registry
export IMAGE_NAME=ttl.sh/spin-app-$(uuidgen):1h
spin registry push $IMAGE_NAME

# Scaffold Kubernetes manifests
spin kube scaffold -f $IMAGE_NAME > app.yaml

# Deploy to Kubernetes
kubectl apply -f app.yaml
```

Reply:
done!